src/jimp2/wireworld/z8/datamangment/DataManager.java

```diff
@@ -19,19 +28,77 @@ public void writeIterationToFile(int iteration, World world, List<Element> eleme
         // TODO implement here
     }
 
-    private Element interpretInputPiece(String input) {
-        // TODO implement here
+    private Element interpretInputPiece(JSONObject jsonObject, String name) {
+        int x1,x2,y1,y2;
+        int width, height;
+        Point point;
+        switch (name) {
+            case "Wire":
+                x1 = ((Long) jsonObject.get("x1")).intValue();
+                x2 = ((Long) jsonObject.get("x2")).intValue();
+                y1 = ((Long) jsonObject.get("y1")).intValue();
+                y2 = ((Long) jsonObject.get("y2")).intValue();
+                Point posiotion1 = new Point(x1, y1);
+                Point posiotion2 = new Point(x2, y2);
+                return new Wire(posiotion1, posiotion2);
+            case "Electron":
+                x1 = ((Long) jsonObject.get("x")).intValue();
+                y1 = ((Long) jsonObject.get("y")).intValue();
+                point = new Point(x1, y1);
+                String orientation = (String) jsonObject.get("orientation");
+                Orientation orient = null;
+                for (Orientation o : Orientation.values()) {
+                    if (o.toString().equals(orientation))
+                        orient = o;
+                }
```

Comment:
Nie zadziałąłoby przypadkiem uproszczenie tego do postaci Orientation.valueOf(orientation); ? Zamiast tej całej pętli i przypisania nulla. Nie robiłem tego do tej pory więc nie jestem 100% pewien ale myśłę że warto przetestować w ten sposób.

Ogólnie jest pewna niekonsekwencja bo Orientation deklarujesz wewnątrz odpowiedniego case'a, a x2, y2, width, height już poza - sugeruję wybranie tylko jednej opcji. Najrozsądniej byłoby chyba zostawić x1, y1 poza switchem (bo to jedyne dane uniwersalne dla każdego elementu) a resztę wpakować do środka, do każdego przypadku indywidualnie.